src/artifact/decl.rs

```diff
@@ -0,0 +1,296 @@
+use crate::artifact::ArtifactError;
+use failure::Error;
+
+/// The kind of declaration this is
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum Decl {
+    /// An import of a function/routine defined in a shared library
+    FunctionImport,
+    /// A GOT-based import of data defined in a shared library
+    DataImport,
+    /// A function defined in this artifact
+    Function { global: bool },
+    /// A data object defined in this artifact
+    Data { global: bool, writable: bool },
```

Comment:
It would, yes. I mostly didn't do that in this PR to avoid breaking existing code, which isnt the best reason since the next few PRs will.

I think a bunch of this code is going to move around a bit in #67 and subsequent, so I'll put that in my mind as a transformation to make in those cases.